### PR TITLE
Improved slice documentation

### PR DESCRIPTION
**Pull request by spencer3035**
Improve slice documentation to include assert_eq checks for all the cases where there were existing examples. I think it makes things more clear when the documentation explicitly checks against values and shows the reader what it does. 

I also started a rust internals discussion about it here: https://internals.rust-lang.org/t/improve-slice-documentaion/21168

### CODE REVIEW

File: library/core/src/slice/iter.rs [hunk 1]
```diff
@@ -914,7 +920,10 @@ impl<T, P> FusedIterator for SplitInclusiveMut<'_, T, P> where P: FnMut(&T) -> b
 ///
 /// ```
 /// let slice = [11, 22, 33, 0, 44, 55];
-/// let iter = slice.rsplit(|num| *num == 0);
+/// let mut iter = slice.rsplit(|num| *num == 0);
+/// assert_eq!(iter.next(), Some(&[44, 55][..]));
+/// assert_eq!(iter.next(), Some(&[10, 22, 33][..]));
```
Comment (jhpratt) [inline library/core/src/slice/iter.rs:925]: Without cloning locally, I believe the issue is that 10 should be 11.

**Review by jhpratt — COMMENTED** [review]
@bors r-

Failing CI. Let me know when it's fixed (please keep it as a single commit) and I'll re-approve.

@rustbot claim

**Comment by spencer3035** [comment]
Classic off by one error. Should be fixed now. @jhpratt

**Comment by jhpratt** [comment]
Can you squash it into one commit?

**Comment by spencer3035** [comment]
I force pushed the two commits into one commit.

**Comment by jhpratt** [comment]
Thanks!

@bors r+

**Pull request merged**